### DISCUSSION TITLE
CMake '3.28.1' was not found in SDK, PATH, or by cmake.dir property.

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1979] [Q&A] [answered]
So, I was trying to setup axmol in my new mac, though I have installed cmake version 3.29.5 through homebrew, I'm getting this error on Android Studio. How do I fix it?

**Comment by rh101** [comment]
Which version of Axmol?

Have you searched for references to "3.28.1" in your project, specifically in `*.gradle` files? Check the `<project>/proj.android/app/build.gradle`, the call to `axutils.findNativeBuildTools`.

That version of cmake is not one referenced in the Axmol source, nor is it a version that is downloaded in the Android Studio SDK tools etc., so it is most likely something your project that is hard-coded to use it.

**Reply by meehjeevan** [reply]
No, I don't have anything hard-coded in my project. 
Axmol is latest from git and I created a blank new project. 
With the same setup it's working on my old imac but now the problem bit weird..  cmake is installed through brew. 
When I run 'which cmake' it throws the default location: /opt/homebrew/bin/cmake which is an alias. 
I tried to set the original location path in .zshrc which has to be "/opt/homebrew/Cellar/cmake/3.29.3/bin" but it's not taking any effect even after sourcing updated .zshrc and even restarting several times. 
I'm using m3 mac now in older mac this issue wasn't occurring.

**Reply by rh101** [reply]
In the `proj.android/app/build.gradle`, change:

`def nbtInfo = axutils.findNativeBuildTools(project)`
to
`def nbtInfo = axutils.findNativeBuildTools(project, cmakeVer = "3.28.1+")`

See if that works.

**Reply by meehjeevan** [reply]
it's already like this:
`def nbtInfo = axutils.findNativeBuildTools(project, '3.28.1+')
`
I think the main problem is cmake is running from alias and it's not getting changed even after changing it to original path in .zshrc it's really weird and frustrating.

**Comment by GamesStudioTwo** [comment]
I use android studio, Click settings, SDK TOOL's  check the Cmake box to install cmake  "Make A Note the version  that it will be installing"   then go to your program directory open your  CMakeLists.txt and change the  cmake_minimum_required(VERSION 3.22) to the version android installed.  Should be good to go. Best of luck.

<img width="530" alt="Sdk" src="https://github.com/axmolengine/axmol/assets/134940573/d4a959c3-28a0-4b32-8174-2203767c67c1">



<img width="697" alt="Cmak" src="https://github.com/axmolengine/axmol/assets/134940573/ea6ee79c-c8f6-4812-a984-f3c343737b9c">

**Reply by meehjeevan** [reply, marked as answer]
ok, I uninstalled cmake from home brew and installed manually, put the path in .zshrc and now it's working.

**Reply by halx99** [reply]
I also have checked install cmake via brew, works

**Reply by crazyhappygame** [reply]
I hit this problem as well.
There was PR created https://github.com/axmolengine/axmol/pull/1987 to change 3.23 to 3.22 (but was not merged...)
Currently we do not need newer for Android newer CMake version than 3.22.
